### ISSUE TITLE
[Bug] Videos are not complete when an exception is thrown

### ISSUE THREAD

**Issue by chaganiu** [microsoft/playwright-java#432]
I have a script like the following:

```java 
try {
    playwright = Playwright.create();
    var launchOptions = new BrowserType.LaunchOptions();
    launchOptions
            .setHeadless(false)
            .setSlowMo(300);
    browser = playwright.chromium().launch(launchOptions);
    var videoPath = Paths.get("videos/");

    var contextOptions = new Browser.NewContextOptions()
            .setViewportSize(1920, 1080)
            .setRecordVideoDir(videoPath)
            .setRecordVideoSize(1920,1080);

    context = browser.newContext(contextOptions);
    page = context.newPage();
    page.setDefaultTimeout(60000);

    // do stuff in the web that eventually causes an exception (`Frame.waitForSelector failed` for example)
} catch (Exception e) {
    e.printStackTrace();
} finally {
    page.close();
    context.close();
    browser.close();
    playwright.close();
}
```

The video is recorded and saved just fine but when i compare where the script failed and the recorded video, the video is missing the end of the script before it failed.

**Comment by yury-s** [comment]
I can reproduce it. The problem happens when the page is explicitly closed, in that case context.close() will not wait for the video to finish recording for that page. I'll prepare a fix.

**Comment by yury-s** [comment]
The fix is now available in 1.12.0-SNAPSHOT and will be released with 1.11.1 too.
